testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs

```diff
@@ -56,47 +56,111 @@ public static void DeleteBuild()
             }
         }
 
-        /// <summary>
-        /// Needs a separate build than the standalone test builds since we don't want the player to try to connect to the editor to do test
-        /// reporting. We only want to main node to do that, worker nodes should be dumb
-        /// </summary>
-        /// <returns></returns>
-        private static BuildReport BuildPlayer(bool isDebug = false)
+        private static BuildReport BuildPlayerUtility(BuildTarget buildTarget = BuildTarget.NoTarget, string buildPathExtension = null, bool buildDebug = false)
         {
-            // Save standalone build path to file so we can read it from standalone tests (that are not running from editor)
-            SaveBuildInfo(new BuildInfo() { BuildPath = BuildPath, IsDebug = isDebug });
+            SaveBuildInfo(new BuildInfo() { BuildPath = BuildPath });
 
             // deleting so we don't end up testing on outdated builds if there's a build failure
             DeleteBuild();
 
+            if (buildTarget == BuildTarget.NoTarget)
+            {
+                if (Application.platform == RuntimePlatform.WindowsPlayer || Application.platform == RuntimePlatform.WindowsEditor)
+                {
+                    buildPathExtension += ".exe";
+                    buildTarget = BuildTarget.StandaloneWindows64;
+                }
+                else if (Application.platform == RuntimePlatform.OSXPlayer ||
+                    Application.platform == RuntimePlatform.OSXEditor)
+                {
+                    buildPathExtension += ".app";
+                    buildTarget = BuildTarget.StandaloneOSX;
+                }
+                else if (Application.platform == RuntimePlatform.LinuxEditor ||
+                    Application.platform == RuntimePlatform.LinuxPlayer)
+                {
+                    buildPathExtension += "";
+                    buildTarget = BuildTarget.StandaloneLinux64;
+                }
```

Comment:
do we really need to take care of extensions manually? wouldn't Unity automatically add extension to path for us?
(also nit line-endings)
```suggestion
                if (Application.platform == RuntimePlatform.WindowsPlayer || Application.platform == RuntimePlatform.WindowsEditor)
                {
                    buildPathExtension += ".exe";
                    buildTarget = BuildTarget.StandaloneWindows64;
                }
                else if (Application.platform == RuntimePlatform.OSXPlayer || Application.platform == RuntimePlatform.OSXEditor)
                {
                    buildPathExtension += ".app";
                    buildTarget = BuildTarget.StandaloneOSX;
                }
                else if (Application.platform == RuntimePlatform.LinuxEditor || Application.platform == RuntimePlatform.LinuxPlayer)
                {
                    buildPathExtension += "";
                    buildTarget = BuildTarget.StandaloneLinux64;
                }
```